quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/InstrumentClass.java

```diff
@@ -277,6 +317,28 @@ public void visitEnd() {
         super.visitEnd();
     }
 
+    private boolean compareSuspendableArgs(Type[] expectedMethodTypes, MethodNode mns) {
+        final Type[] candidateMethodTypes = Type.getArgumentTypes(mns.desc);
+        for (int i = 0; i < candidateMethodTypes.length; i++) {
+            if (!Objects.equals(candidateMethodTypes[i], expectedMethodTypes[i + 1])) {
```

Comment:
I think it is worth a comment on this function illustrating what the signatures of the method signatures look like when they "match".  I had an impression in my head of what I thought they looked like which doesn't align with this, but then realised I don't really know.

e.g. 
`fun foo$default(arg0, arg1)`
`fun foo(<mask>, arg0, arg1)`

Reply:
Added additional comment